tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,737 @@
+        boolean missingRequiredValue = false;
+        if (descriptor.getInputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockInputLocation(count));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getOutputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockOutputLocation(count));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getFileSystemRealmName() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockFilesystemRealmName(count));
+            missingRequiredValue = true;
+        }
+        if(descriptor.getHashCharset() == null) {
+            descriptor.setHashCharset(StandardCharsets.UTF_8);
+        }
+        if(descriptor.getHashEncoding() == null) {
+            descriptor.setHashEncoding(Encoding.BASE64);
+        }
+        if(descriptor.getEncoded() == null) {
+            descriptor.setEncoded(true);
+        }
+        if(descriptor.getLevels() == 0) {
+            descriptor.setLevels(2);
+        }
+        if (missingRequiredValue) {
+            descriptor.reset();
+        }
+    }
+
+    /**
+     * Handles creating the Elytron filesystem-realm from the descriptor array
+     *
+     * @throws Exception Exception to be handled by Elytron Tool
+     */
+    private void createFileSystemRealm() throws Exception {
+        Security.addProvider(new WildFlyElytronProvider());
+        for (Descriptor descriptor : descriptors) {
+            System.out.println("Creating encrypted realm for: " + descriptor.getInputRealmLocation());
+            if (descriptor.getFileSystemRealmName() == null || descriptor.getInputRealmLocation() == null || descriptor.getOutputRealmLocation() == null) {
+                continue;
+            }
+            CredentialStore credentialStore;
+            // check if credential-store and secret-key-alias are both specified, or both null
+            if (! (descriptor.getCredentialStore() == null ^ descriptor.getSecretKeyAlias() == null) ){
+                boolean isNew = false;
+                if (descriptor.getCredentialStore() == null) {
+                    descriptor.setCredentialStore(Paths.get(descriptor.getOutputRealmLocation(), "mycredstore.cs").toString());
+                    descriptor.setSecretKeyAlias("key");
+                    isNew = true;
+                }
+                String csType = PropertiesCredentialStore.NAME;
+                try {
+                    credentialStore = CredentialStore.getInstance(csType);
+                } catch (NoSuchAlgorithmException e) {
+                    // fallback to load all possible providers
+                    credentialStore = CredentialStore.getInstance(csType, getProvidersSupplier(null));
+                }
+                Map<String, String> implProps = new HashMap<>();
+                implProps.put("create", "true");
```

Comment:
I think this should be a parameter value. See `CREATE_CREDENTIAL_STORE_PARAM = "create"` in `CredentialStoreCommand`.

Reply:
Added parameter